packages/core/src/indexing-store/utils/sort.ts

```diff
@@ -42,7 +43,8 @@ export function buildOrderByConditions({
           (columnName) =>
             isScalarColumn(table[columnName]) ||
             isReferenceColumn(table[columnName]) ||
-            isEnumColumn(table[columnName]),
+            isEnumColumn(table[columnName]) ||
+            isJSONColumn(table[column]),
```

Comment:
I think we probably want to disallow sorting by JSON columns (just like filtering).